CMakeLists.txt

```diff
@@ -538,6 +532,12 @@ source_group("Source Files" FILES ${SRC_FILES})
 source_group("Header Files" FILES ${API_HEADER_FILES} ${INC_FILES})
 set(ALL_SOURCE_FILES ${SRC_FILES} ${API_HEADER_FILES} ${INC_FILES})
 
+# Shorten the source file pathing for log messages
+foreach(SRC_FILE ${SRC_FILES})
+  string(REPLACE "${CMAKE_SOURCE_DIR}/" "" LOG_FILE_ ${SRC_FILE})
+  set_source_files_properties(${SRC_FILE} PROPERTIES COMPILE_FLAGS "-DLOG_FILE_='\"${LOG_FILE_}\"'")
```

Comment:
@trygvis In order to properly define  the `LOG_FILE_` preprocessor directive for the source files could you please change `"-DLOG_FILE_='\"${LOG_FILE_}\"'"` to `-DLOG_FILE_=\\\"${LOG_FILE_}\\\"`. This change is required for Windows to build properly; otherwise the preprocessor directive is treated as a character constant [see [C2015](https://msdn.microsoft.com/en-us/library/s7f33t4h.aspx)]. Verified this change on Windows (VS2013) and Linux (Ubuntu 14.04).

Feel free to squash the commits. Other than that, thank you for this fix; was a much needed correction.